lib/mobility/plugins/upload_for_translation.rb

```diff
@@ -11,61 +14,23 @@ def apply(attributes, option)
         end
       end
 
-      UPLOAD_TRANSLATION_DELAY = (ENV["UPLOAD_FOR_TRANSLATION_DELAY_MIN"] || 5).minutes
-
       def write(locale, value, options = {})
-        if should_upload_for_translation?(model, locale, value, attribute)
-          # Get translated attributes, and each attribute's params for uploading translations
-          translated_attribute_names = model_class.translated_attribute_names
-          translated_attribute_params = {}
-          translated_attribute_names.each do |attribute_name|
-            upload_options = model.public_send("#{attribute_name}_backend").options[:upload_for_translation]
-            translated_attribute_params[attribute_name] = upload_options.is_a?(Hash) ? upload_options : {}
-          end
-
-          # Asynchronously upload attributes for translations
-          # Include a delay so that multiple edits to the same object can be 'de-duped' in the async job.
-          I18nSonder::Workers::UploadSourceStringsWorker.perform_in(
-              UPLOAD_TRANSLATION_DELAY,
-              model.class.name,
-              model[:id],
-              {
-                translated_attribute_params: translated_attribute_params,
-                namespace: namespace
-              }
-          )
-        end
+        return unless should_upload_for_translation?(model, locale, value, attribute)
+
+        # Asynchronously upload attributes for translations
```

Comment:
Since most of this seems to be common with the `UploadSourceStrings` class (especially if we add the `should_upload_for_translation?` check, could we simplify all this in a similar way to the `SyncTranslationsWorker` as follows:

```
module I18nSonder
  module UploadSourceStrings
    def upload
       ...
  end
end

module Plugins
  module UploadForTranslation
    include I18nSonder::UploadSourceStrings

      def write
        upload
  end
end
```

and forego the UploadHelpers class?

Reply:
Yeah that makes sense